librz/core/cbin.c

```diff
@@ -694,8 +694,32 @@ static void add_map(RzCore *core, RZ_NULLABLE RzCoreFile *cf, RzBinMap *map, ut6
 		return;
 	}
 
-	// then we map the part of the section that comes from the physical file
-	char *map_name = map->name ? rz_str_newf("fmap.%s", map->name) : rz_str_newf("fmap.%d", fd);
+	const char *prefix = "fmap";
+
+	// open and use a different fd for virtual files
+	if (map->vfile_name) {
+		char *uri = rz_str_newf("vfile://%" PFMT32u "/%s", bf->id, map->vfile_name);
```

Comment:
How will this work for projects? Is `bf->id` here ok for serializing/deserializing projects?
BTW, do we want to have projects considered immediately in each PR or as an afterthought? That is, shall we require you/contributors to immediately fix projects when they do these kind of changes or is it ok to have them fixed "later"?

Reply:
In this case, since fine-grained bin/io serialization is not implemented at all yet, this does not need and updates in projects.

> How will this work for projects? Is bf->id here ok for serializing/deserializing projects?

In the way that I plan to implement bin/io serialization, yes this will be ok. I plan to make the ids and fds stable across loads. But even if they would not be stable (for example if we want to deserialize io/bin loadings on top of already loaded stuff, not sure if useful) then the loading code can reassign new ids as long as the correspondences are correctly represented in the serialization.

> BTW, do we want to have projects considered immediately in each PR or as an afterthought? That is, shall we require you/contributors to immediately fix projects when they do these kind of changes or is it ok to have them fixed "later"?

When projects are stable, they will have to be considered and updated/fixed immediately per commit (not per pr, if the pr is not squashed). Since we are in beta right now we can see it a little bit relaxed but should still update immediately whenever possible.